syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package transport_catalogue_serialize;

message Coordinates {
    double latitude = 1;
    double longitude = 2;
}

message Stop {
    uint32 id = 1;
    string name = 2;
    Coordinates coordinates = 3;
}

message Bus {
    string name = 1;
    repeated uint32 stops = 2;
    bool is_roundtrip = 3;
}

message Distance {
    uint32 start_stop_id = 1;
    uint32 end_stop_id = 2;
    double distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
}

message AggregatedData {
    TransportCatalogue transport_catalogue = 1;
    render_serialize.RenderSettings render_settings = 2;
    transport_router_serialize.RoutingSettings routing_settings = 3;
}